build_tools/azure/test_docs.sh

```diff
@@ -5,6 +5,17 @@ set -ex
 source build_tools/shared.sh
 activate_environment
 
-# XXX: for some unknown reason python -m pytest fails here in the CI, can't
-# reproduce locally and not worth spending time on this
-pytest $(find doc -name '*.rst' | sort)
+scipy_doctest_installed=$(python -c 'import scipy_doctest' && echo "True" || echo "False")
+if [[ "$scipy_doctest_installed" == "True" ]]; then
+    doc_rst_files=$(find $PWD/doc -name '*.rst' | sort)
+    # Changing dir, as we do in build_tools/azure/test_script.sh, avoids an
+    # error when importing sklearn. Not sure why this happens ... I am going to
+    # wild guess that it has something to do with the bespoke way we set up
+    # conda with putting conda in the PATH and source activate, rather than
+    # source <conda_root>/etc/profile.d/conda.sh + conda activate.
+    cd $TEST_DIR
+    # with scipy-doctest, --doctest-modules only runs doctests (in contrary to
+    # vanilla pytest where it runs doctests on top of normal tests)
+    python -m pytest --doctest-modules --pyargs sklearn
+    python -m pytest --doctest-modules $doc_rst_files
```

Comment:
I'm not sure about the last one, rst files. In scipy we use a slightly different incantation:

```
$ pytest --doctest-glob=*rst ../repos/scipy/scipy/doc/source/tutorial
```

Whether these two are equivalent or not, I've no idea.

Reply:
Not sure either, I guess ours is listing explicit files through bash, and the other one is relying on pytest functionality for finding all the files to doctest.

I guess if it's not broken don't fix it :wink: